modules/storage-api/src/main/java/org/apache/ignite/internal/storage/DataStorageModules.java

```diff
@@ -112,12 +104,50 @@ public Map<String, StorageEngine> createStorageEngines(
     /**
      * Collects {@link DataStorageConfigurationSchema data storage schema} fields (with {@link Value}).
      *
+     * @param polymorphicSchemaExtensions {@link PolymorphicConfigInstance Polymorphic schema extensions} that contain extensions of {@link
+     * DataStorageConfigurationSchema}.
      * @return Mapping: {@link DataStorageModule#name Data storage name} -> filed name -> field type.
+     * @throws IllegalStateException If the {@link DataStorageConfigurationSchema data storage schemas} are not valid.
      */
-    public Map<String, Map<String, Class<?>>> collectSchemasFields() {
+    public Map<String, Map<String, Class<?>>> collectSchemasFields(Collection<Class<?>> polymorphicSchemaExtensions) {
+        Map<String, Class<? extends DataStorageConfigurationSchema>> schemas = polymorphicSchemaExtensions.stream()
+                .filter(DataStorageConfigurationSchema.class::isAssignableFrom)
+                .collect(toUnmodifiableMap(
+                        schemaCls -> schemaName((Class<? extends DataStorageConfigurationSchema>) schemaCls),
+                        schemaCls -> (Class<? extends DataStorageConfigurationSchema>) schemaCls
+                ));
+
+        Set<String> dataStorageNames = modules.stream()
+                .map(DataStorageModule::name)
+                .collect(toSet());
+
+        if (!dataStorageNames.equals(schemas.keySet())) {
+            List<String> dataStorageWithoutSchema = dataStorageNames.stream()
```

Comment:
Please extract content of this "if" into a method. It is self-contained and pretty big

Reply:
Fix it.